test/controllers/role_skill_controller_test.exs

```diff
@@ -1,148 +1,88 @@
 defmodule CodeCorps.RoleSkillControllerTest do
-  use CodeCorps.ApiCase
-
-  alias CodeCorps.RoleSkill
-  alias CodeCorps.Repo
-
-  @valid_attrs %{}
-  @invalid_attrs %{}
-
-  defp build_payload, do: %{ "data" => %{"type" => "role-skill", "attributes" => %{}}}
+  use CodeCorps.ApiCase, resource_name: :role_skill
 
   describe "index" do
     test "lists all entries on index", %{conn: conn} do
-      path = conn |> role_skill_path(:index)
-      json = conn |> get(path) |> json_response(200)
-      assert json["data"] == []
+      [role_skill_1, role_skill_2] = insert_pair(:role_skill)
+
+       conn
+       |> request_index
+       |> json_response(200)
+       |> assert_ids_from_response([role_skill_1.id, role_skill_2.id])
     end
 
     test "filters resources on index", %{conn: conn} do
-      elixir = insert(:skill, title: "Elixir")
-      phoenix = insert(:skill, title: "Phoenix")
-      rails = insert(:skill, title: "Rails")
-
-      role = insert(:role)
-      role_skill_1 = insert(:role_skill, role: role, skill: elixir)
-      role_skill_2 = insert(:role_skill, role: role, skill: phoenix)
-      insert(:role_skill, role: role, skill: rails)
+      [role_skill_1, role_skill_2] = insert_pair(:role_skill)
```

Comment:
You figured out the missing third record, so I'll just link https://github.com/code-corps/code-corps-api/pull/425/files#r86388037 as a pointer on how to write this in one line.